django/db/models/constraints.py

```diff
@@ -54,8 +54,9 @@ def __repr__(self):
         return "<%s: check='%s' name=%r>" % (self.__class__.__name__, self.check, self.name)
 
     def __eq__(self, other):
+        if not isinstance(other, CheckConstraint):
+            return NotImplemented
```

Comment:
`CheckConstraint`, `UniqueConstraint` and `ExclusionConstraint` inherit form `BaseConstraint` it should be fine to call `super().__eq__(other)` if other's class doesn't match, e.g.
```python
def __eq__(self, other):
    if isinstance(other, UniqueConstraint):
        return (
            self.name == other.name and
            self.fields == other.fields and
            self.condition == other.condition
        )
    return super().__eq__(other)
```